### PR TITLE
sponge support.

### PR DESCRIPTION
**Pull request by portlek**
not tested*
#21

### CODE REVIEW

**Comment by Cubxity** [comment]
Would it be possible for you to test this out?

File: sponge/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+repositories {
+    maven("https://repo.spongepowered.org/maven")
+}
+
+dependencies {
+    api(project(":unifiedmetrics-core"))
+    compileOnly("org.spongepowered", "spongeapi", "7.3.0")
+    annotationProcessor("org.spongepowered:spongeapi:7.3.0")
```
Comment (Cubxity) [inline sponge/build.gradle.kts:30]: kapt should be used here

File: sponge/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+sponge.plugin.id="umetrics"
```
Comment (Cubxity) [inline sponge/build.gradle.kts:33]: please format this

File: sponge/src/main/kotlin/dev/cubxity/plugins/metrics/sponge/metric/EventsMetric.kt [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    override val collectors: List<MetricCollector> =
+        listOf(joinCounter, quitCounter, chatCounter, pingCounter)
+
+    override fun initialize() {
+        Sponge.getEventManager().registerListeners(bootstrap.plugin, this)
+    }
+
+    override fun dispose() {
+        Sponge.getEventManager().unregisterListeners(bootstrap.plugin)
```
Comment (Cubxity) [inline sponge/src/main/kotlin/dev/cubxity/plugins/metrics/sponge/metric/EventsMetric.kt:46]: This should only unregister listeners associated with this class

**Review by Cubxity — CHANGES_REQUESTED** [review]

File: sponge/src/main/kotlin/dev/cubxity/plugins/metrics/sponge/bootstrap/UMSpongePlugin.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+import java.nio.file.Path;
+
+@Plugin(
+        id = "umetrics",
+        name = "UnifiedMetrics",
+        version = "0.2.1",
+        description = "Fully-featured metrics plugin for Minecraft servers"
+)
+public class UMSpongePlugin {
```
Comment (Cubxity) [inline sponge/src/main/kotlin/dev/cubxity/plugins/metrics/sponge/bootstrap/UMSpongePlugin.java:39]: Why is this class written in Java?

**Review by Cubxity — CHANGES_REQUESTED** [review]

**Comment by portlek** [comment]
I couldn't run the plugin in the sponge, when the sponge is reading the plugin, the server shuts down instantly and I made it in java but nothing change, I should work on the p.r.

**Pull request closed without merging**